src/device/router_device_worker.erl

```diff
@@ -2075,10 +2081,39 @@ save_and_update(DB, CF, Pid, Device) ->
     ok = router_device_channels_worker:handle_device_update(Pid, Device).
 
 -spec get_device(rocksdb:db_handle(), rocksdb:cf_handle(), binary()) -> router_device:device().
-get_device(DB, CF, ID) ->
-    case router_device:get_by_id(DB, CF, ID) of
-        {ok, D} -> D;
-        _ -> router_device:new(ID)
+get_device(DB, CF, DeviceID) ->
+    Device =
+        case router_device:get_by_id(DB, CF, DeviceID) of
+            {ok, D} -> D;
+            _ -> router_device:new(DeviceID)
+        end,
+    case router_console_api:get_device(DeviceID) of
+        {error, not_found} ->
+            lager:info("device not found"),
+            Device;
+        {error, _Reason} ->
+            lager:error("failed to get device ~p", [_Reason]),
+            Device;
+        {ok, APIDevice} ->
+            lager:info("got device: ~p", [APIDevice]),
+            RxDelayOld = maps:get(rx_delay, router_device:metadata(Device), 0),
+            RxDelayNew = maps:get(rx_delay, router_device:metadata(APIDevice), RxDelayOld),
+            Metadata =
+                case RxDelayOld == RxDelayNew of
+                    true ->
+                        router_device:metadata(APIDevice);
+                    false ->
+                        maps:put(new_rx_delay, RxDelayNew, router_device:metadata(APIDevice))
+                end,
```

Comment:
Depending upon which gets merged first:

This expression for populating `Metadata` and where it's used below will create a merge conflict if `dpezely/fix-rx_delay-for-join-accept` gets merged to master first.

In that case, remove this entire `Metadata` statement and variable, and where it gets used: that expression gets replaced by a single `maps:merge()` call.